### ISSUE TITLE
Parsing of JSON Decimals as Ruby Floats Can Result In Loss of Precision

### ISSUE THREAD

**Issue by jaschob** [ruby/json#219]
When JSON decimals are parsed, they are converted to Ruby floats, which can result in a loss of precision. This is of course a known limitation of floating point representation but here's an example to illustrate:

```
irb(main):001:0> require 'json'
=> false
irb(main):002:0> JSON::VERSION
=> "1.8.1"
irb(main):003:0> JSON('{ "bd": 100200300400500.01 }')["bd"]
=> 100200300400500.02
```

I realize there's support for customizing the parsing of JSON objects into any Ruby class using the `:create_additions` option, but in cases where I can't control the JSON being generated, I have no recourse. As a specific example, the bitcoind RPC interface returns amounts simply as JSON numbers, and there's no way to have it return a JSON object (like `{ "json_class": "BigDecimal", "b": "3.141" }`) instead.

I'm submitting this more as a question than an actual issue report. I am wondering:
1. Is there a way to parse JSON floats into Ruby without losing precision?
2. If not, would there be any interest in an additional parsing option (say `:decimal_class => BigDecimal`) that allows the caller to designate an alternate class to use for JSON floats?

**Comment by soundasleep** [comment]
Has this issue been resolved in #223 ?

**Comment by byroot** [comment]
> Has this issue been resolved in #223 ?

I think so.
